### PR TITLE
Update compute_stats_by_span_kind documentation

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?

Updates documentation for compute_stats_by_span_kind based on changes from https://github.com/DataDog/datadog-agent/pull/22163. Also adds clarification in previous release note to address similar config options.

### Motivation

Clarification on top-level spans update.

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -1553,8 +1553,8 @@ api_key:
   ## @env DD_APM_COMPUTE_STATS_BY_SPAN_KIND - bool - default: false
   ## [BETA] Enables an additional stats computation check on spans to see they have an eligible `span.kind` (server, consumer, client, producer).
   ## If enabled, a span with an eligible `span.kind` will have stats computed. If disabled, only top-level and measured spans will have stats computed.
-  ## NOTE: For stats computed from OTel traces, only top-level spans are considered when this option is off.
-  ## If you are sending OTel traces and want stats on non-top-level spans, this flag will need to be enabled.
+  ## NOTE: Stats are now computed by span kind for OTel traces by default.
```
Comment (maycmlee) [inline pkg/config/config_template.yaml:1556]:
```suggestion
  ## NOTE: Stats are computed by span kind for OTel traces by default.
```

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -1553,8 +1553,8 @@ api_key:
...
+  ## If you are sending OTel traces and do not want stats computed by span kind, this flag will need to be disabled and the "disable_otlp_compute_top_level_by_span_kind" APM Feature will need to be enabled.
```
Comment (maycmlee) [inline pkg/config/config_template.yaml:1557]:
```suggestion
  ## If you are sending OTel traces and do not want stats computed by span kind, you need to disable this flag and enable the "disable_otlp_compute_top_level_by_span_kind" APM feature.
```

File: releasenotes/notes/otel-top-level-spans-3e391e77dd7ebf83.yaml [hunk 3]
```diff
@@ -11,3 +11,4 @@ features:
     OTLP ingest now identifies top-level spans by span kind for OpenTelemetry spans by default. This is both a breaking change and a bug fix that may increase the number of spans that generate trace metrics. This new logic can be disabled if needed by adding `disable_otlp_compute_top_level_by_span_kind` in DD_APM_FEATURES.
     - With this new logic, root spans and spans with a server or consumer `span.kind` will be marked as top-level. Additionally, spans with a client or producer `span.kind` will have stats computed.
     - If `disable_otlp_compute_top_level_by_span_kind` is enabled, this new logic will be disabled and OpenTelemetry spans may be misidentified as top-level.
+    - Please note that in order to disable computing stats by span kind for OTel traces, `disable_otlp_compute_top_level_by_span_kind` needs to be enabled and `apm_config.compute_stats_by_span_kind` needs to be disabled.
```
Comment (maycmlee) [inline releasenotes/notes/otel-top-level-spans-3e391e77dd7ebf83.yaml:14]:
```suggestion
    - To disable computing stats by span kind for OTel traces, enable `disable_otlp_compute_top_level_by_span_kind` and disable `apm_config.compute_stats_by_span_kind`.
```

**Review by maycmlee — COMMENTED** [review]
Left a couple of suggestions!

**Review by maycmlee — APPROVED** [review]

**Pull request merged**